usb.go

```diff
@@ -165,7 +165,27 @@ func newContextWithImpl(impl libusbIntf) *Context {
 
 // NewContext returns a new Context instance.
 func NewContext() *Context {
-	return newContextWithImpl(libusbImpl{})
+	return ContextOptions{}.New()
+}
+
+// DeviceDiscovery enables a scan of the available USB devices connected to the system
+type DeviceDiscovery int
+
+const (
+	EnableDeviceDiscovery = iota
+	DisableDeviceDiscovery
+)
+
+// ContextOptions holds optional flags checked when creating a new Context
```

Comment:
"optional flags" is too specific, since we might add more parameters to the context creation that do not easily translate into options in libusb init.

Perhaps:
ContextOptions holds parameters for Context initialization.

Reply:
Good point. I always need to remind myself - gousb is _not_ a libusb wrapper. Not _only_ a wrapper at least.

Reply:
Anything still missing here?